client/src/pages/AddRide.js

```diff
@@ -8,59 +8,111 @@ import Card from "../components/Card";
 import HeaderMain from "../components/HeaderMain";
 import Input from "../components/Input";
 import Button from "../components/Button";
+import Todo from "../components/Todo";
 
 const PageWrapper = styled.div`
-  position: fixed;
-  height: 100%;
+  min-height: 100%;
   width: 100%;
-  background: var(--text-secondary);
-  overflow: auto;
-`;
+  margin: auto;
+  padding: 8rem 0;
 
-const ContentWrapper = styled.div`
-  padding-top: 12rem;
+  background: var(--text-secondary);
 `;
-
-const Form = styled.form`
+const Wrapper = styled.div`
   display: flex;
   flex-direction: column;
   align-items: center;
   margin: 2rem auto;
-  width: 80vw;
+  padding: 0 1rem;
+  max-width: 400px;
+  > :first-child {
+    margin-bottom: 2rem;
+  }
+`;
+const Form = styled.form`
   > * {
     margin-top: 0.7rem;
   }
   > :first-child {
     margin-top: 0;
   }
-  > :last-child {
-    margin-top: 1.5rem;
+`;
+const StyledDetails = styled.details`
+  color: var(--text-primary);
+  text-align: left;
+  margin: 1rem 0;
+  width: 100%;
+  textarea {
+    width: 100%;
   }
-  > div {
-    display: flex;
-    > * {
-      margin-right: 0.5rem;
-    }
-    > :last-child {
-      margin-right: 0;
-    }
+  h3 {
+    margin-top: 1rem;
   }
 `;
 
+const OutputContainer = styled.div`
+  margin-top: 0.25rem;
+  margin-left: 0;
+`;
+const CheckboxEntry = styled.div`
+  display: flex;
+  button {
+    margin-left: 1rem;
+    padding: 0 1rem;
+  }
+`;
+const InputContainer = styled.form`
+  color: var(--text-primary);
+  display: flex;
+  align-items: center;
+
+  input[type="checkbox"] {
+    margin-right: 1rem;
+  }
+  input[type="text"] {
+    height: 2rem;
+    width: 100%;
+  }
+  button {
+    height: 2rem;
+    margin: auto 1rem;
+    padding: 0 1rem;
+  }
+`;
+
+const BadgeContainer = styled.div`
+  margin: 1rem auto;
+
+  display: flex;
+  > * {
+    margin-right: 0.5rem;
+  }
+  > :last-child {
+    margin-right: 0;
+  }
+`;
 export default function AddRide() {
   const [title, setTitle] = useState("");
   const [start, setStart] = useState("");
   const [dest, setDest] = useState("");
   const [date, setDate] = useState("");
   const [rider, setRider] = useState("");
   const [priority, setPriority] = useState("normal");
-  const [cargo, setCargo] = useState("s");
+  const [cargo, setCargo] = useState(null);
+  // const [info, setInfo] = useState("");
+  const [checkbox, setCheckbox] = useState("");
+  const [checkboxes, setCheckboxes] = useState([]);
   const history = useHistory();
 
+  const handleRemove = (entry) => {
+    const filteredCheckboxes = checkboxes.filter((cb) => cb !== entry);
+    setCheckboxes(filteredCheckboxes);
+  };
```

Comment:
This is based on content rather than unique identification

Reply:
I already noticed ... should i add an index or uuid to fix it?
otherwise there wont be a checkbox with exactly the same content in the end